### PR TITLE
[Performance] Fix usages of $interval

### PR DESCRIPTION
**Pull request by VWoeltjen**
This updates usages of $interval to set `invokeApply` to `false` when desired, to reduce number of digests that get triggered. Addresses #369 

This is most significant in MCTSplitPane; without this change, there are 140 digests/sec while idle. With this change, that number drops below 10 (accounted for by indicators etc which do refresh once per second.)

Summary of changes:
* Updated digest indicator (from example/profiling) to give more current results
* Fixed usage of $interval in MCTSplitPane
* Added spec for MCTSplitPane in order to verify usage of $interval
  * Spec remains mostly empty for now; WTD-1400 exists to backfill these
* Use `invokeApply=false` from `mct-chart` to reduce watch counts associated with plotting
* Fixed usage of $interval for the ElasticSearch indicator
  * This does still trigger watches as a consequence of its $http usages, but these currently occur at 15-second intervals

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Expect to pass code review? Y

### CODE REVIEW

**Comment by VWoeltjen** [comment]
Reassigning to self to investigate CircleCI failure

File: platform/commonUI/general/src/directives/MCTSplitPane.js [hunk 1]
```diff
@@ -204,7 +204,7 @@ define(
                 // And poll for position changes enforced by styles
                 activeInterval = $interval(function () {
                     getSetPosition(getSetPosition());
-                }, POLLING_INTERVAL, false);
+                }, POLLING_INTERVAL, 0, false);
```
Comment (akhenry) [inline platform/commonUI/general/src/directives/MCTSplitPane.js]: Wow, nice catch :+1:

**Comment by akhenry** [comment]
This looks good.

One question though, it's out of scope for this issue, but couldn't the resize stuff be event driven, rather than on a polling cycle?

**Comment by larkin** [comment]
I think event driven is an ideal way to do it-- but as I recall there are some complications that prevent it from working when race conditions occur in css loading.

Prefer to defer that improvement until after API discussions.

Pete

On Dec 3, 2015, at 6:18 PM, akhenry <notifications@github.com<mailto:notifications@github.com>> wrote:


This looks good.

One question though, it's out of scope for this issue, but couldn't the resize stuff be event driven, rather than on a polling cycle?

**Comment by VWoeltjen** [comment]
The reason it's on a polling cycle is because the size (before the user starts moving it) is dependent on CSS, which in turn is dependent on its location in the DOM. It's sensitive to a lot of details, including changes to parent classes (in addition to the timing of CSS loading.) Unfortunately the events we'd need to listen to just don't exist (although we could probably come up with an 80% solution with some documented caveats.)

**Comment by akhenry** [comment]
Fair enough, I figured there must be good reasons. Thanks for the explanation.

**Pull request merged**